crates/typst-ide/src/complete.rs

```diff
@@ -908,8 +908,10 @@ fn complete_code(ctx: &mut CompletionContext) -> bool {
         return false;
     }
 
-    // An existing identifier: "{ pa| }".
-    if ctx.leaf.kind() == SyntaxKind::Ident {
+    // An existing identifier (which is not the key of a named pair): "{ pa| }".
+    if ctx.leaf.kind() == SyntaxKind::Ident
+        && (ctx.leaf.index() > 0 || ctx.leaf.parent_kind() != Some(SyntaxKind::Named))
+    {
```

Comment:
An attempt at being a bit more descriptive:

```suggestion
    // An existing identifier: "{ pa| }".
    // Ignores named pair keys as they are not variables (as in "func(pa|: 23)").
    if ctx.leaf.kind() == SyntaxKind::Ident
        && (ctx.leaf.index() > 0 || ctx.leaf.parent_kind() != Some(SyntaxKind::Named))
    {
```